### PR TITLE
[19.03] gitlab 12.1.6 -> 2.2.3

### PR DESCRIPTION
**Pull request by flokli**
##### Motivation for this change
19.03 backport for https://github.com/NixOS/nixpkgs/pull/67770/commits

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
 
-  version = "8.7.0";
+  version = "8.8.1";
 
   src = fetchFromGitLab {
     owner = "gitlab-org";
     repo = "gitlab-workhorse";
     rev = "v${version}";
-    sha256 = "1zlngc498hnzbxwdjn3ymr0xwrnfgnzzhn9lyf37yfbjl8x28n3z";
+    sha256 = "02a1z2nky31acd17q2n4pi4jkbkr6x6frfi6qlcsmfza0x0kzpc0";
```
Comment (jonringer) [inline pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix]: for darwin, it looks like there's a different release of this:
```
hash mismatch in fixed-output derivation '/nix/store/kmrla6v72mv9gjk1pm9nwi7mlnsl4qis-source':
  wanted: sha256:02a1z2nky31acd17q2n4pi4jkbkr6x6frfi6qlcsmfza0x0kzpc0
  got:    sha256:0dfzib24a23z92jw2h3pzbl1d7m56rbf36l55mblwhx7nhan9v2g
```

Comment (flokli) [reply pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix]: Huh? @LnL7, any thoughts?

Comment (flokli) [reply pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix]: Fetching from gitlab should be the same on both darwin and linux. Maybe the used git binary behaves differently?

Comment (jonringer) [reply pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix]: very odd, i know, but that's what ofborg is tripping on

Comment (jonringer) [reply pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix]: doing a nix-review on my macbook yields spectacular failures all over the place from trying to build ruby packages :(

Comment (flokli) [reply pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix]: @jonringer did it previously work on macos?

Comment (jonringer) [reply pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix]: looks like that's a no.... gitlab only supports linux. Now I'm curious as to why ofborg tried to instantiate it...

**Comment by flokli** [comment]
Let's close this until https://github.com/NixOS/nixpkgs/pull/67770 is ready.

I restricted `meta.platforms` to `platforms.linux` on all gitlab server components and pushed a rebased version there.

**Pull request closed without merging**